File: Backend/servers/subscriptions/src/ping/views/no_authentication.py
from fastapi import status
from fastapi.responses import JSONResponse
from pydantic import BaseModel


async def get_not_auth():
    response = {"message": "Ping only not authorisation level from server"}
    return JSONResponse(content=response, status_code=status.HTTP_200_OK)


class Data(BaseModel):
    message: str


async def post_not_auth(data: Data):
    message_1 = "Ping only not authorisation level from server"
    message_2 = "Request data: " + data.message
    response = {message_1: message_1, message_2: message_2}
    return JSONResponse(content=response, status_code=status.HTTP_200_OK)

File: Backend/servers/11 recommendations/src/ping/routes/no_authentication.rs
use crate::ping::datatypes::{DualMessage, Message};
use actix_web::{web::Json, HttpResponse, Responder, Result};

/// A get route function that requires the user to be unauthenticated
pub async fn ping_get_not_auth() -> Result<impl Responder> {
    let message: Message = Message {
        message: String::from("Ping only not authorized level from server"),
    };

    return Ok(HttpResponse::Ok()
        .content_type("application/json; charset=utf-8")
        .json(message));
}

/// A post route function that requires the user to be unauthenticated
pub async fn ping_post_not_auth(data: Json<Message>) -> Result<impl Responder> {
    let req_data: String = data.into_inner().message;

    let message_1: String = String::from("Ping only not authorized level from server");
    let message_2: String = format!("Request data: {}", req_data);

    let dual_message: DualMessage = DualMessage {
        message_1,
        message_2,
    };
    println!("dual_message: {:?}", dual_message);
    return Ok(HttpResponse::Ok()
        .content_type("application/json; charset=utf-8")
        .json(dual_message));
}